src/queue.rs

```diff
@@ -0,0 +1,98 @@
+use core::{
+    marker::PhantomData,
+    mem::{size_of, MaybeUninit},
+};
+
+use esp_idf_sys::TickType_t;
+
+use crate::sys;
+
+pub struct Queue<'a, T> {
```

Comment:
Is it worth it to have this lifetime here...? Maybe not, as it's only use is the borrowing of the queue in the UART driver, which can be done in a different way (see below). I suggest you remove this lifetime, as it would improve the queue ergonomics when used standalone.